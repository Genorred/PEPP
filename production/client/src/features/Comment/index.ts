export { default as PostComment } from "./ui/PostComment";
export { default as CommentForm } from "./ui/CommentForm";
