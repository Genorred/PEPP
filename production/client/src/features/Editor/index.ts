export { EditPost } from "./ui/EditPost";
export { default as SaveWork } from "./ui/SaveWork";
export { usePostEditor } from "./lib/usePostEditor";
export { editor } from "./consts/editor";
