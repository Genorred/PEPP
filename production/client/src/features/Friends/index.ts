export { default as Friends } from "./ui/Friends";
export { default as IncomingRequests } from "./ui/IncomingRequests";
