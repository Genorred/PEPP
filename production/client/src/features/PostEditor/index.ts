export { editor } from './consts/editor'