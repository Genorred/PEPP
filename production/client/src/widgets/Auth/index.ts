export { default as AuthWrapper } from "./ui/AuthWrapper";
export { default as AuthForm } from "./ui/AuthForm";
