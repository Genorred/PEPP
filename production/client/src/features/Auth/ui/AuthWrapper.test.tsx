import { fireEvent, render, screen } from "@testing-library/react";
import { AuthWrapper } from "@/features/Auth";
import { useRouter } from "next/navigation";

jest.mock("next/navigation");

it("Auth wrapper handles google button", () => {
  (useRouter as jest.Mock).mockReturnValue({
    push: jest.fn(),
  });

  const returnUrl = "http://localhost:5678/auth/google";

  const { asFragment } = render(<AuthWrapper returnUrl={returnUrl} children={<></>} />);
  expect(screen.getByText(/Sign in With Google/i));
  expect(asFragment).toMatchSnapshot();

  const button = screen.getByTestId("sign-in-button");
  fireEvent.click(button);

  const url = new URL("http://localhost:7878/auth/google");
  url.searchParams.set("returnUrl", returnUrl);
  expect(useRouter().push).toHaveBeenCalledWith(url.href);
});