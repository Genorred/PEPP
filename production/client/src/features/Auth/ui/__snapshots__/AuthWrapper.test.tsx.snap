// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth wrapper handles google button 1`] = `[Function]`;
